plugins/woocommerce/tests/php/src/Internal/DataStores/Orders/OrdersTableDataStoreTests.php

```diff
@@ -1015,6 +1007,149 @@ public function test_get_order_id_by_order_key() {
 		$this->assertEquals( 0, $this->sut->get_order_id_by_order_key( 'other_order_key' ) );
 	}
 
+	/**
+	 * @testDox Direct write to metadata should propagate to the orders table when reading.
+	 */
+	public function test_read_with_direct_meta_write() {
+		$this->enable_cot_sync();
+		$order = $this->create_complex_cot_order();
+
+		$post_object = get_post( $order->get_id() );
+		assert( get_post_type( $post_object->ID ) === 'shop_order' );
```

Comment:
Curious: why a native assertion and not a PHPUnit assertion?

Reply:
I am using native here, because if this is failing then its likely because of some other reason which is not related to the code that this UT is testing. So I wanted to throw an error rather than cause a test failure. Let me know if this makes sense.

Reply:
I'm a big fan of making the code blow up as loudly as possible on failures that are caused by it doing something wrong, but maybe this approach should be implemented in the real code instead; I think unit tests should contain only test assertions, which will caught our attention anyway if they fail.

As a side note, `assert` is an interesting function to be used in code, since it gets executed or not depending on how the PHP engine is configured: https://www.php.net/manual/en/function.assert.php

Reply:
Addressed here https://github.com/woocommerce/woocommerce/pull/34465/commits/5bdb847e0b2758ee8a1e3f2da6a5356ac9f3e5a4